pkg/assembler/backends/testing/hasSLSA.go

```diff
@@ -98,65 +146,56 @@ func (c *demoClient) registerHasSLSA(selectedPackage *model.Package, selectedSou
 		newHasSlsa.Subject = selectedArtifact
 	}
 
-	for _, pack := range builtFromPackages {
-		newHasSlsa.BuiltFrom = append(newHasSlsa.BuiltFrom, pack)
-	}
-	for _, source := range builtFromSouces {
-		newHasSlsa.BuiltFrom = append(newHasSlsa.BuiltFrom, source)
-	}
-	for _, art := range builtFromArtifacts {
-		newHasSlsa.BuiltFrom = append(newHasSlsa.BuiltFrom, art)
-	}
-
 	c.hasSLSA = append(c.hasSLSA, newHasSlsa)
 	return nil
 }
 
 // Query HasSlsa
 
 func (c *demoClient) HasSlsa(ctx context.Context, hasSLSASpec *model.HasSLSASpec) ([]*model.HasSlsa, error) {
-
-	if hasSLSASpec.Package != nil && hasSLSASpec.Source != nil && hasSLSASpec.Artifact != nil {
-		return nil, gqlerror.Errorf("cannot specify package, source and artifact together for hasSLSASpec")
+	subjectsDefined := 0
+	if hasSLSASpec.Subject.Package != nil {
+		subjectsDefined = subjectsDefined + 1
 	}
-	if hasSLSASpec.Package != nil && hasSLSASpec.Source != nil {
-		return nil, gqlerror.Errorf("cannot specify package and source together for hasSLSASpec")
+	if hasSLSASpec.Subject.Source != nil {
+		subjectsDefined = subjectsDefined + 1
 	}
-	if hasSLSASpec.Package != nil && hasSLSASpec.Artifact != nil {
-		return nil, gqlerror.Errorf("cannot specify package and artifact together for hasSLSASpec")
+	if hasSLSASpec.Subject.Artifact != nil {
+		subjectsDefined = subjectsDefined + 1
 	}
-	if hasSLSASpec.Source != nil && hasSLSASpec.Artifact != nil {
-		return nil, gqlerror.Errorf("cannot specify source and artifact together for hasSLSASpec")
+	if subjectsDefined > 1 {
+		return nil, gqlerror.Errorf("Must specify at most one subject (package, source, or artifact)")
 	}
```

Comment:
hmm, does the 0 case need to be handled as well?

Reply:
For the query if the subject is not defined, you query all

Reply:
Right, but will uncover these cases on the e2e tests and can fix there.

Reply:
I don't think there's an issue here: 0 means match everything and it should match (testing e2e pending to uncover other bugs), 1 means match exactly one of them